model/src/main/java/com/cloudera/nav/sdk/model/IdAttrs.java

```diff
@@ -0,0 +1,81 @@
+// (c) Copyright 2015 Cloudera, Inc. All rights reserved.
+package com.cloudera.nav.sdk.model;
+
+import com.cloudera.nav.sdk.model.annotations.MProperty;
+import com.cloudera.nav.sdk.model.entities.EntityType;
+import com.cloudera.nav.sdk.model.entities.TagChangeSet;
+import com.cloudera.nav.sdk.model.entities.UDPChangeSet;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.Maps;
+
+import java.util.Collection;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.joda.time.Instant;
+
+/**
+ * Used for specifying the attributes needed for generation of entity id's at
+ * the server side.
+ */
+public class IdAttrs {
```

Comment:
Can we revisit doing this on the client side? The more I look at this the more it seems too dirty from a design perspective. There must be a better way than essentially blowing up the abstractions we've built around the different Entity classes.

Reply:
have removed this class altogether since we don't need this anymore